.github/workflows/check_issue_label.yml

```diff
@@ -0,0 +1,115 @@
+            }
+            // ======================
+            // TEAM MEMBERSHIP CHECK
+            // ======================
+            async function getTeamMembers() {
+              try {
+                const result = await octokit.graphql({
+                  query: `query ($org: String!, $teamSlug: String!) {
+                    organization(login: $org) {
+                      team(slug: $teamSlug) {
+                        members(first: 100) {
+                          nodes { login }
+                        }
+                      }
+                    }
+                  }`,
+                  org: 'oppia',
+                  teamSlug: 'oppia-good-first-issue-labelers'
+                });
+                return new Set(
+                  result.organization.team.members.nodes
+                    .map(node => node.login.toLowerCase())
+                );
+              } catch (error) {
+                console.error('Team membership check failed:', error);
+                return new Set(); // Fail closed
+              }
+            }
+            // ======================
+            // LABEL ENFORCEMENT
+            // ======================
+            async function enforceLabelRules() {
+              const authorizedMembers = await getTeamMembers();
+              const prSpecificLabels = new Set(['dependencies', 'stale']);
+              // Handle 'good first issue' label
+              if (sanitizedLabel.toLowerCase() === 'good first issue') {
+                if (!authorizedMembers.has(actorUsername)) {
+                  await octokit.rest.issues.removeLabel({
+                    owner,
+                    repo,
+                    issue_number: issueNumber,
+                    name: sanitizedLabel
+                  });
+                  await octokit.rest.issues.createComment({
+                    owner,
+                    repo,
+                    issue_number: issueNumber,
+                    body: `@${actorUsername}, only [oppia-good-first-issue-labelers team members](https://github.com/orgs/oppia/teams/oppia-good-first-issue-labelers) can apply this label.`
+                  });
+                }
+                return;
+              }
+              // Block PR-specific labels
+              if (prSpecificLabels.has(sanitizedLabel.toLowerCase()) || 
+                  sanitizedLabel.startsWith('PR:')) {
+                await octokit.rest.issues.removeLabel({
+                  owner,
+                  repo,
+                  issue_number: issueNumber,
+                  name: sanitizedLabel
+                });
+                await octokit.rest.issues.createComment({
+                  owner,
+                  repo,
+                  issue_number: issueNumber,
+                  body: `@${actorUsername}, the ${sanitizedLabel} label is reserved for pull requests.`
+                });
+              }
+            }
+            await enforceLabelRules();
```

Comment:
as I mentioned in https://github.com/oppia/oppia/pull/21828#discussion_r1942176417, we should send failure notifications for this workflow